lib/applications/resolvers.ts

```diff
@@ -64,3 +68,161 @@ export const createApplication: Resolver = async (_, args, { prisma }) => {
+    throw new UpdatedFieldsMissingError('Missing updated personal address fields');
+  }
+
+  // Validate updated contact info fields (at least one of phone or email must be provided)
+  if (updatedContactInfo && !phone && !email) {
+    throw new UpdatedFieldsMissingError('Missing updated contact info fields');
+  }
+
+  // Validate updated doctor info fields
+  if (
+    updatedPhysician &&
+    (!physicianFirstName ||
+      !physicianLastName ||
+      !physicianMspNumber ||
+      !physicianAddressLine1 ||
+      !physicianCity ||
+      !physicianPostalCode)
+  ) {
+    throw new UpdatedFieldsMissingError('Missing updated physician fields');
+  }
+
+  // Retrieve applicant record
+  const applicant = await prisma.applicant.findUnique({
+    where: { id: applicantId },
+    include: { medicalInformation: { include: { physician: true } } },
+  });
+
+  // If applicant not found, throw error
+  if (!applicant) {
+    throw new ApplicantNotFoundError(`No applicant with ID ${applicantId} was found`);
+  }
+
+  const physician = applicant.medicalInformation?.physician;
+
+  // Temporary Shopify confirmation number placeholder
+  // TODO: Integrate with Shopify payments
+  const currentDateTime = new Date().getTime().toString();
+  const shopifyConfirmationNumber = currentDateTime.substr(currentDateTime.length - 7);
+
+  let application;
+  try {
+    application = await prisma.application.create({
+      data: {
+        firstName: applicant.firstName,
+        lastName: applicant.lastName,
+        dateOfBirth: applicant.dateOfBirth,
+        gender: applicant.gender,
+        customGender: applicant.customGender,
+        email: updatedContactInfo && email ? email : applicant.email,
+        phone: updatedContactInfo && phone ? formatPhoneNumber(phone) : applicant.phone,
+        province: applicant.province,
+        city: updatedAddress && city ? city : applicant.city,
+        addressLine1: updatedAddress && addressLine1 ? addressLine1 : applicant.addressLine1,
+        addressLine2: updatedAddress && addressLine2 ? addressLine2 : applicant.addressLine2,
+        postalCode:
+          updatedAddress && postalCode ? formatPostalCode(postalCode) : applicant.postalCode,
+        isRenewal: true,
+        // TODO: Link with Shopify checkout
+        shopifyConfirmationNumber,
+        processingFee: 26,
+        paymentMethod: PaymentType.Cash,
+        // TODO: Modify logic when DB schema gets changed (medicalInfo is not undefined)
+        disability: applicant.medicalInformation?.disability || 'Placeholder disability',
+        physicianFirstName:
+          updatedPhysician && physicianFirstName
+            ? physicianFirstName
+            : physician?.firstName || 'Placeholder physician first name',
+        physicianLastName:
+          updatedPhysician && physicianLastName
+            ? physicianLastName
+            : physician?.lastName || 'Placeholder physician last name',
```

Comment:
NOTE: This is going to change with the change that combines physician's firstName and lastName into one name field.